unstable/lens/hash-slice.rkt

```diff
@@ -0,0 +1,27 @@
+#lang racket/base
+
+(provide hash-pluck-lens)
+
+(require racket/list
+         lens/base/main
+         "hash.rkt"
+         "compound.rkt")
+(module+ test
+  (require rackunit))
+
+(define (hash-pluck-lens . ks)
+  (apply compound-hash-lens
+         (append-map
+          (λ (k)
+            (list k (hash-ref-lens k)))
+          ks)))
+
+(module+ test
+  (check-equal? (lens-view (hash-pluck-lens 'a 'c) (hash 'a 1 'b 2 'c 3))
+                (hash 'a 1 'c 3))
+  (check-equal? (lens-set (hash-pluck-lens 'a 'c) (hash 'a 1 'b 2 'c 3) (hash 'a 4 'c 5))
+                (hash 'a 4 'b 2 'c 5))
+  (test-case "this one currently fails"
+    (check-equal? (lens-set (hash-pluck-lens 'a 'c) (hash 'a 1 'b 2 'c 3) (hash 'a 4))
+                  (hash 'a 4 'b 2 'c 3)))
```

Comment:
It doesn't, no. If you do a `set (hash 'a 4)`, then when you do a view you should get `(hash 'a 4)` but instead you'll get `(hash 'a 4 'c 3)` - the set isn't equal to the view. Specifically, this test fails:

```racket
(check-set-view (hash-pluck-lens 'a 'c)
                          (hash 'a 1 'b 2 'c 3)
                          (hash 'a 100))
```

Fix is to either make the set remove the 'c key and have the view not include keys that aren't in the hash or to not allow setting the view to a hash with fewer keys